ariadne_codegen/plugins/shorter_results.py

```diff
@@ -0,0 +1,294 @@
+            method_def.returns.id, self.class_dict
+        )
+        if node_and_class is None:
+            return super().generate_client_method(method_def)
+
+        return_node, single_field_class, single_field_return_class = node_and_class
+
+        # Update the return type to be the inner field and ensure we reference
+        # the single field before returning.
+        method_def.returns = return_node
+        method_def.body[-1] = generate_return(
+            value=generate_attribute(
+                value=return_stmt.value, attr=single_field_return_class
+            )
+        )
+
+        self._update_imports(method_def, single_field_class)
+
+        return super().generate_client_method(method_def)
+
+    def _update_imports(self, method_def, single_field_class):
+        if single_field_class not in self.class_dict:
+            return
+
+        # After we change the type we also need to import it in the client if
+        # it's one of our generated types so add the extra import as needed.
+        if method_def.name not in self.extended_imports:
+            self.extended_imports[method_def.name] = set()
+
+        self.extended_imports[method_def.name].add(single_field_class)
+
+
+def _return_or_yield_node_and_class(
+    current_return_class: str, class_dict: Dict[str, ast.ClassDef]
+) -> Optional[tuple[ast.expr, str, str]]:
+    if current_return_class not in class_dict:
+        return None
+
+    return_class_ast = class_dict[current_return_class]
+
+    # If we're not a single method class there's nothing for the plugin to do.
+    if len(return_class_ast.body) != 1:
+        return None
+
+    # If we are a single method class, ensure we inherit from BaseModel. If we
+    # inherit from another class like a fragment that could contain more fields
+    # so we don't want to unpack.
+    if len(return_class_ast.bases) != 1:
+        return None
+
+    if not isinstance(return_class_ast.bases[0], ast.Name):
+        return None
+
+    if not return_class_ast.bases[0].id == "BaseModel":
+        return None
+
+    single_field = return_class_ast.body[0]
+    if single_field is None:
+        return None
+
+    if not isinstance(single_field, ast.AnnAssign):
+        return None
+
+    if not isinstance(single_field.target, ast.Name):
+        return None
+
+    # Traverse the type annotation until we find the inner type. This can
+    # require several iterations if the return type is something like
+    # Optional[List[Any]]
+    return_node, return_class = _update_node(single_field.annotation)
+
+    return return_node, return_class, single_field.target.id
+
+
+def _update_node(node: ast.expr) -> tuple[ast.expr, str]:
+    """
+    Recurse down a node to finner the inner ast.Name. Once found, evaluate the
+    potential literal so it gets unquoted and return the inner name.
+    """
+    if isinstance(node, ast.Name):
+        try:
+            node.id = ast.literal_eval(node.id)
```

Comment:
This will also update fields annotation, either we should "rewrite" whole annotation with modified values or use original one without changing it. For example (from test you added) `ListTypeAListOptionalTypeA` is no longer quoted:
```py
list_optional_type_a: List[Optional[ListTypeAListOptionalTypeA]] = Field(
        alias="listOptionalTypeA"
    )
```

Reply:
Now making a copy of the annotation and updating the one we're using in the client but keeping it quoted for the original annotation.